src/Storages/StorageURL.cpp

```diff
@@ -1490,23 +1513,44 @@ FormatSettings StorageURL::getFormatSettingsFromArgs(const StorageFactory::Argum
     return format_settings;
 }
 
+std::string evaluateBodySubqueryToString(ASTPtr body_ptr, const std::optional<String>& format, const ContextPtr & context)
+{
+    auto write_buffer = WriteBufferFromOwnString();
+
+    BlockIO block = interpretSubquery(body_ptr, context, {}, {})->execute();
```

Comment:
It uses old planner infrastructure, it's better to evaluate it like this https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/StorageExecutable.cpp#L183